### ISSUE TITLE
Problem deploying OVN Controllers

### ISSUE THREAD

**Issue by flyersa** [osism/issues#653]
Hello,

i am struggling a bit to setup OVN for a testbed installation. It is refusing to deploy the ovn-controller always saying there are no hosts found:

```
PLAY [Apply role ovn-controller] 
skipping: no hosts matched
```
network, compute and control roles are set accordingly. If i override ovn-controller:children then it will rollout the controller, but also only to the control nodes, it is refusing to rollout anything to compute, even if specified. And then it will create them as gateway controller instead of control controller.

I checked some other environments with ovn, there are nowhere any overrides or custom roles set for the ovn deployment.

Basically i run openvswitch deployment first, then ovn. Am i missing something here? Since we did all other rollouts before with ovs i never encountered this issue before (obviously).

configuration.yml:

enable_ovn: "yes"
enable_openvswitch: "yes"
neutron_plugin_agent: "ovn"
neutron_ovn_distributed_fip: "yes"
enable_neutron_agent_ha: "yes"
enable_neutron_dvr: "yes"
enable_neutron_fwaas: "no"
enable_neutron_qos: "yes"
enable_neutron_vpnaas: "no"

ovn_northd, sb_db and nb_db are deployed fine on all controllers. Im just missing controllers, thus i have no network agents and neutron metadata agents complains it cant find the chassis in the databases.

**Comment by osfrickler** [comment]
I think `enable_ovn` and `enable_openvswitch` are mutually exclusive, you should only enable one? The OVS part that is needed for OVN should be included automatically. And `enable_neutron_agent_ha` should also only be set for an OVS deployment?

But then, the testbed by default only has AIO node, how did you modify the setup to get compute-only nodes?

**Comment by flyersa** [comment]
Testbed is not the osism testbed, its a testbed installation by us.

Doesnt really matter if i remove those settings, i just tried if they fix something (which they dont), other environments with ovn do not have them at all (the enable part). So also removing them, or only let enable_ovn: yes doesnt change anything. for the ha_agent settings also, that doesnt effect any deployment part of ovn in regards to the controllers

**Comment by berendt** [comment]
Please share inventory/10-custom, inventory/20-roles, and inventory/99-overwrite.

**Comment by flyersa** [comment]
any idea?

**Comment by berendt** [comment]
Not yet, no. Do you use the ovn play or the ovn-controller play? It looks like there is a bug in the ovn-controller play.  But we always use the ovn play everywhere.

**Comment by berendt** [comment]
container-image-kolla-ansible#504 should fix the problem in the ovn play when having the ovn-controller on different nodes.

**Comment by flyersa** [comment]
how can i test this? should i manually add those in the container?

**Comment by berendt** [comment]
Yes. You can edit the ``kolla-ovn.yml`` file inside the kolla-ansible container.

**Comment by flyersa** [comment]
ok that looks alot better now.

```
(.venv-osism-install) root@open-mgmt:~# openstack --os-cloud admin network agent list
+-------------+------------------------------+-------------+-------------------+-------+-------+----------------+
| ID          | Agent Type                   | Host        | Availability Zone | Alive | State | Binary         |
+-------------+------------------------------+-------------+-------------------+-------+-------+----------------+
| open-cmpt01 | OVN Controller Gateway agent | open-cmpt01 |                   | :-)   | UP    | ovn-controller |
| open-cmpt02 | OVN Controller Gateway agent | open-cmpt02 |                   | :-)   | UP    | ovn-controller |
+-------------+------------------------------+-------------+-------------------+-------+-------+----------------+

```

but should i not also have ovn controller controller agents on the ctrls?

**Comment by berendt** [comment]
That's the kolla-ansible default:

```
[ovn-controller:children]
ovn-controller-compute
ovn-controller-network

[ovn-controller-compute:children]
compute

[ovn-controller-network:children]
network
```

**Comment by flyersa** [comment]
Well im stil not sure why the controllers do not roll out "controller controller agents", but im also lacking the time to dig deep into that. We will roll out this testbed now with OVS which we know works flawless, and will create a own testsetup to try to replicate the problem we faced. 

While with your change ovn deploys at least as Gateway Controllers it is refusing to create ports (getting mechanism driver failures as example). We also wiped everything already and started full over 2 times now, nothing fixes that mechanism driver error on port creation and i cant find meaningfull error logs outside of (port xy not found).

Maybe close this for now. But thanks alot for this fix, at least this is adresses now. I will try to replicate in our own lab with a new installation and may reopen
